File: Android/ScantrustConsumerSDK/app/src/main/java/com/scantrust/dev/android/scantrustconsumersdk/data/model/ScanResultResponse.kt
package com.scantrust.dev.android.scantrustconsumersdk.data.model

data class ScanResultResponse(
    val code: CodeInfo,
    val campaign: CampaignInfo,
    val scan: ScanInfo
)

data class CodeInfo(
    val product: ProductInfo,
    val brand: BrandInfo,
    val qrcode: QRCodeInfo,
    val scm_data: List<SCMData>,
    val scan_count: Int
)

data class ProductInfo(
    val id: Int,
    val name: String,
    val image: String,
    val description: String,
    val sku: String,
    val client_url: String
)

data class BrandInfo(
    val id: Int,
    val name: String,
    val description: String,
    val image: String
)

data class QRCodeInfo(
    val message: String,
    val creation_date: String,
    val serial_number: String,
    val is_blacklisted: Boolean,
    val activation_status: String,
    val blacklist_reason: String,
    val is_consumed: Boolean
)

data class SCMData(
    val name: String,
    val key: String,
    val position: Int,
    val type: String,
    val value: String
)

data class CampaignInfo(
    val products: List<ProductInfo>,
    val name: String,
    val options: CampaignOptions
)

data class CampaignOptions(
    val is_enabled: Boolean,
    val serial_number_lookup_enabled: Boolean,
    val complaints: ComplaintsConfig,
    val wechat: WeChatConfig,
    val mark_consumed: Boolean,
    val ga_key: String,
    val stc_config: Map<String, Any>
)

data class ComplaintsConfig(
    val is_enabled: Boolean,
    val to_email: String
)

data class WeChatConfig(
    val is_enabled: Boolean,
    val channel_redirect: ChannelRedirect
)

data class ChannelRedirect(
    val is_enabled: Boolean,
    val landing_page: String,
    val channel_id: String
)

data class ScanInfo(
    val app: String,
    val reason: String,
    val result: String,
    val auth_failure_mode: String,
    val country: String
)

File: iOS/ScantrustConsumerSDK/ScantrustConsumerSDK/Models/ScanResultResponse.swift
//
//  ScanResultResponse.swift
//  ScantrustConsumerSDK
//
//  Created by AI Assistant on 2024-06-09.
//

import Foundation

struct ScanResultResponse: Codable {
    let code: CodeInfo
    let campaign: Campaign
    let scan: ScanInfo
}

struct CodeInfo: Codable {
    let product: Product
    let brand: Brand
    let qrcode: QRCode
    let scmData: [SCMData]
    let scanCount: Int

    enum CodingKeys: String, CodingKey {
        case product
        case brand
        case qrcode
        case scmData = "scm_data"
        case scanCount = "scan_count"
    }
}

struct Product: Codable {
    let id: Int
    let name: String
    let image: String?
    let description: String?
    let sku: String?
    let clientUrl: String?

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case image
        case description
        case sku
        case clientUrl = "client_url"
    }
}

struct Brand: Codable {
    let id: Int
    let name: String
    let description: String?
    let image: String?
}

struct QRCode: Codable {
    let message: String
    let creationDate: String
    let serialNumber: String?
    let isBlacklisted: Bool
    let activationStatus: String
    let blacklistReason: String?
    let isConsumed: Bool

    enum CodingKeys: String, CodingKey {
        case message
        case creationDate = "creation_date"
        case serialNumber = "serial_number"
        case isBlacklisted = "is_blacklisted"
        case activationStatus = "activation_status"
        case blacklistReason = "blacklist_reason"
        case isConsumed = "is_consumed"
    }
}

struct SCMData: Codable {
    let name: String
    let key: String
    let position: Int
    let type: String
    let value: String
}

struct Campaign: Codable {
    let products: [Product]
    let name: String
    let options: CampaignOptions
}

struct CampaignOptions: Codable {
    let isEnabled: Bool
    let serialNumberLookupEnabled: Bool

    enum CodingKeys: String, CodingKey {
        case isEnabled = "is_enabled"
        case serialNumberLookupEnabled = "serial_number_lookup_enabled"
    }
}

struct ScanInfo: Codable {
    let app: String
    let reason: String
    let result: String
    let authFailureMode: String
    let country: String

    enum CodingKeys: String, CodingKey {
        case app
        case reason
        case result
        case authFailureMode = "auth_failure_mode"
        case country
    }
}